assets/dev/js/editor/components/selection/manager.js

```diff
@@ -0,0 +1,212 @@
+export default class Manager extends elementorModules.editor.utils.Module {
+	/**
+	 * Selected elements.
+	 *
+	 * The list of the selected elements.
+	 *
+	 * @type {{}}
+	 */
+	elements = {};
+
+	/**
+	 * Selected elements type.
+	 *
+	 * Represents the common type of multiple selected elements, or false when the selected elements are of different
+	 * types.
+	 *
+	 * @type {string|boolean}
+	 */
+	type = false;
+
+	/**
+	 * Manager constructor.
+	 *
+	 * @returns {Manager}
+	 */
+	constructor() {
+		super();
+
+		return new Proxy( this, {
+			get: function( target, prop ) {
+				// Add a hook after executing the `add`/`remove` methods.
+				if ( [ 'add', 'remove' ].includes( prop ) ) {
+					return ( ...args ) => {
+						const result = target[ prop ]( ...args );
+
+						target.updateType();
+						target.updatePanelPage();
+						target.updateNavigator();
+
+						return result;
+					};
+				}
+
+				return Reflect.get( ...arguments );
+			},
+		} );
```

Comment:
Why not calling another 3rd method after each `add` \ `remove` ends to handle this logic, instead of proxy?

Reply:
Add comment about the usage of the Proxy